sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/_tests_/student-loan-balances-queue.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,110 @@
+import { DeepMocked, createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, getISODateOnlyString } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { mockDownloadFiles } from "@sims/test-utils/mocks";
+import * as Client from "ssh2-sftp-client";
+import { Job } from "bull";
+import * as path from "path";
+import { Student } from "@sims/sims-db";
+import { StudentLoanBalancesScheduler } from "../student-loan-balances-queue.scheduler";
+
+// Student loan balances received file mocks.
+const STUDENT_LOAN_BALANCES_FILENAME = "PEDU.PBC.PT.MBAL.20240302.001";
+
+describe(describeProcessorRootTest(QueueNames.StudentLoanBalances), () => {
+  let app: INestApplication;
+  let processor: StudentLoanBalancesScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let studentLoanBalancesDownloadFolder: string;
+
+  beforeAll(async () => {
+    studentLoanBalancesDownloadFolder = path.join(
+      __dirname,
+      "student-loan-balances-files",
+    );
+    // Set the ESDC response folder to the files mocks folders.
+    process.env.ESDC_RESPONSE_FOLDER = studentLoanBalancesDownloadFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(StudentLoanBalancesScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it("Should add monthly loan balance record for the student.;", async () => {
+    // Arrange
+    // Create student if it doesn't exist.
+    let student: Student = await db.student.findOne({
+      where: {
+        birthDate: getISODateOnlyString(new Date("1998-03-24")),
+        user: { lastName: "FOUR" },
+        sinValidation: { sin: "900041310" },
+      },
+    });
+    if (!student) {
+      student = await saveFakeStudent(db.dataSource);
+      // Update the student to ensure that the student receiving loan balance is the same student as the one created above.
+      student.birthDate = getISODateOnlyString(new Date("1998-03-24"));
+      student.user.lastName = "FOUR";
+      student.sinValidation.sin = "900041310";
+      await db.student.save(student);
+    }
+    // Queued job.
+    const job = createMock<Job<void>>();
```

Comment:
We have been missing this but we should mock the job as below. The return object also allow access to logs and allows the assertions to happen is a facilitated way.
```ts
const { job } = mockBullJob<Job<void>>();
```
or
```ts
const jobMock = mockBullJob<Job<void>>();
// jobMock.logMessages
// jobMock.job
// jobMock.containLogMessage
```
@andrepestana-aot @lewischen-aot @dheepak-aot @katrina-bc FYI